File: generate.js
#!/usr/bin/env node
// Generates all quiz HTML files from the template and quiz data
const fs = require('fs');
const path = require('path');

const TEMPLATE = fs.readFileSync(path.join(__dirname, 'quiz-template.html'), 'utf8');
const QUIZZES = require('./quiz-data.json');

for (const quiz of QUIZZES) {
  const html = TEMPLATE
    .replace(/\{\{QUIZ_TITLE\}\}/g, quiz.title)
    .replace(/\{\{CX_LIFECYCLE\}\}/g, quiz.cx_lifecycle)
    .replace(/\{\{DIFFICULTY\}\}/g, quiz.difficulty)
    .replace(/\{\{QUESTION_COUNT\}\}/g, String(quiz.questions.length))
    .replace(/\{\{DIFFICULTY_SHORT\}\}/g, quiz.difficulty_short)
    .replace('{{QUIZ_DATA_JSON}}', JSON.stringify({ title: quiz.title, questions: quiz.questions }, null, 2));

  const outPath = path.join(__dirname, quiz.filename);
  fs.writeFileSync(outPath, html);
  console.log(`Created: ${quiz.filename} (${quiz.questions.length} questions)`);
}

console.log(`\nDone! Generated ${QUIZZES.length} quiz files.`);

File: generate.py
#!/usr/bin/env python3
"""Generates all quiz HTML files from the template and quiz data."""
import json
import os

DIR = os.path.dirname(os.path.abspath(__file__))

with open(os.path.join(DIR, 'quiz-template.html'), 'r') as f:
    TEMPLATE = f.read()

with open(os.path.join(DIR, 'quiz-data.json'), 'r') as f:
    QUIZZES = json.load(f)

for quiz in QUIZZES:
    quiz_json = json.dumps({
        'title': quiz['title'],
        'questions': quiz['questions']
    }, indent=2)

    html = TEMPLATE
    html = html.replace('{{QUIZ_TITLE}}', quiz['title'])
    html = html.replace('{{CX_LIFECYCLE}}', quiz['cx_lifecycle'])
    html = html.replace('{{DIFFICULTY}}', quiz['difficulty'])
    html = html.replace('{{QUESTION_COUNT}}', str(len(quiz['questions'])))
    html = html.replace('{{DIFFICULTY_SHORT}}', quiz['difficulty_short'])
    html = html.replace('{{QUIZ_DATA_JSON}}', quiz_json)

    out_path = os.path.join(DIR, quiz['filename'])
    with open(out_path, 'w') as f:
        f.write(html)

    print(f"Created: {quiz['filename']} ({len(quiz['questions'])} questions)")

print(f"\nDone! Generated {len(QUIZZES)} quiz files.")
